flask_mqtt/__init__.py

```diff
@@ -80,11 +81,12 @@ def init_app(self, app):
         # type: (Flask) -> None
         """Init the Flask-MQTT addon."""
         self.client_id = app.config.get("MQTT_CLIENT_ID", "")
-        self.client = Client(
-            client_id=self.client_id,
-            transport=app.config.get("MQTT_TRANSPORT", "tcp"),
-            protocol=app.config.get("MQTT_PROTOCOL_VERSION", MQTTv311)
-        )
+        if isinstance(self.client_id, unicode):
+            self.client._client_id = self.client_id.encode('utf-8')
+        else:
+            self.client._client_id = self.client_id
+        self.client._transport = app.config.get("MQTT_TRANSPORT", "tcp").lower()
```

Comment:
I don't like the lower() function there. The transport settings are defined clearly in the documentation and there should not be multiple options for upper and lower case here.

Reply:
I write this line because this code is executed during __init__ function of Client.
(https://github.com/eclipse/paho.mqtt.python/blob/master/src/paho/mqtt/client.py line 515)
I think is not mandatory and Upper value is not a problem but i try to keep consistency of code with paho MQTT.

Reply:
OK, I see. Pahos __init__ function uses `lower()` here so it makes sense to do it the same way in the `init_app()` function.